internal/osutils/autostart/legacy_darwin.go

```diff
@@ -0,0 +1,33 @@
+package autostart
+
+import (
+	"os"
+
+	"github.com/ActiveState/cli/internal/errs"
+	"github.com/ActiveState/cli/internal/fileutils"
+	"howett.net/plist"
+)
+
+type LegacyPlist struct {
+	ProgramArguments []string `plist:"ProgramArguments"`
+}
+
+func isLegacyPlist(path string) (bool, error) {
+	if !fileutils.FileExists(path) {
+		return false, nil
+	}
+
+	reader, err := os.Open(path)
+	if err != nil {
+		return false, errs.Wrap(err, "Could not open plist file")
+	}
+
+	decoder := plist.NewDecoder(reader)
+	var p LegacyPlist
+	err = decoder.Decode(&p)
+	if err != nil {
+		return false, errs.Wrap(err, "Could not decode plist file")
+	}
```

Comment:
Wouldn't these error conditions mean it's not a plist file? Are we not causing errors here for users who aren't upgrading from this legacy state?

Reply:
Without knowing more about plist files I don't think we can make that assumption. 

For users upgrading who aren't in a legacy state the return on line 32 should be false. We still have the `ProgramArguments` key in the file, which is an array but there should only be one entry in the non-legacy case.